From 8427a4c4099a66ff65b69d9bb5654ed0cb73d620 Mon Sep 17 00:00:00 2001
From: Mohsen Dolaty <mohsen.dolaty@amd.com>
Date: Thu, 29 Sep 2022 11:13:00 -0500
Subject: [PATCH 1/1] linux-aspeed: Turin initial DTS

Add initial device tables for 5 new platform for Turin
Delete device tables for SP5

Signed-off-by: Mohsen Dolaty <mohsen.dolaty@amd.com>
---
 ...-quartz.dts => aspeed-bmc-amd-chalupa.dts} |   0
 arch/arm/boot/dts/aspeed-bmc-amd-galena.dts   | 859 ++++++++++++++++++
 ...titanite.dts => aspeed-bmc-amd-huambo.dts} |   0
 ...amd-ruby.dts => aspeed-bmc-amd-purico.dts} |   0
 ...md-onyx.dts => aspeed-bmc-amd-recluse.dts} |   0
 5 files changed, 859 insertions(+)
 rename arch/arm/boot/dts/{aspeed-bmc-amd-quartz.dts => aspeed-bmc-amd-chalupa.dts} (100%)
 create mode 100644 arch/arm/boot/dts/aspeed-bmc-amd-galena.dts
 rename arch/arm/boot/dts/{aspeed-bmc-amd-titanite.dts => aspeed-bmc-amd-huambo.dts} (100%)
 rename arch/arm/boot/dts/{aspeed-bmc-amd-ruby.dts => aspeed-bmc-amd-purico.dts} (100%)
 rename arch/arm/boot/dts/{aspeed-bmc-amd-onyx.dts => aspeed-bmc-amd-recluse.dts} (100%)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-chalupa.dts
similarity index 100%
rename from arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
rename to arch/arm/boot/dts/aspeed-bmc-amd-chalupa.dts
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-galena.dts b/arch/arm/boot/dts/aspeed-bmc-amd-galena.dts
new file mode 100644
index 000000000000..a142d7a88f2d
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-galena.dts
@@ -0,0 +1,859 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2021 AMD Inc.
+// Author: Sreedhar Govardhana <srgovard@amd.com>
+// Author: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
+
+/dts-v1/;
+
+#include "aspeed-g6.dtsi"
+#include "dt-bindings/gpio/aspeed-gpio.h"
+
+/ {
+	model = "AMD Galena BMC";
+	compatible = "amd,galena-bmc", "aspeed,ast2600";
+
+	aliases {
+		serial4 = &uart5;
+		ethernet0 = &mac3;
+		ethernet1 = &mac2;
+	};
+
+	chosen {
+		stdout-path = &uart5;
+		bootargs = "console=ttyS4,115200 earlycon vmalloc=512MB";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		fault {
+			gpios = <&gpio0 ASPEED_GPIO(O, 6) GPIO_ACTIVE_HIGH>;
+		};
+
+		identify {
+			gpios = <&gpio0 ASPEED_GPIO(O, 7) GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	memory@80000000 {
+		device_type = "memory";
+		reg = <0x80000000 0x80000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		video_engine_memory: jpegbuffer {
+			size = <0x02000000>;	/* 32M */
+			alignment = <0x01000000>;
+			compatible = "shared-dma-pool";
+			reusable;
+		};
+
+		pcc_memory: pccbuffer {
+			no-map;
+			reg = <0xE0000000 0x00001000>; /* 4K */
+		};
+	};
+
+	iio-hwmon-adc121c {
+		compatible = "iio-hwmon";
+		oemname0 = "FPGA_VDD_CORE";
+		oemname1 = "FPGA_VDD_18";
+		oemname2 = "FPGA_VDD_33";
+		oemname3 = "LOM_VDD_33";
+		oemname4 = "LOM_VDD_12";
+		oemname5 = "VDD_12_DUAL";
+		io-channels = <&adc0 0>, <&adc1 0>, <&adc2 0>, <&adc3 0>,
+					<&adc4 0>, <&adc5 0>;
+	};
+};
+
+&mctp0 {
+	status	= "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	ethphy0: ethernet-phy@0 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <0>;
+	};
+};
+
+&mac2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii3_default>;
+	clock-names = "MACCLK",	"RCLK";
+	use-ncsi;
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rgmii";
+	phy-handle = <&ethphy0>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii4_default>;
+};
+
+&fmc {
+	status = "okay";
+	fmc-spi-user-mode;
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		status = "okay";
+		label = "bmc";
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+		#include "openbmc-flash-layout-128.dtsi"
+	};
+};
+
+&spi1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1_default &pinctrl_qspi1_default>;
+	fmc-spi-user-mode;
+	compatible = "aspeed,ast2600-spi";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		label = "pnor";
+		status = "okay";
+	};
+
+};
+
+&spi2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi2_default &pinctrl_qspi2_default>;
+	fmc-spi-user-mode;
+	compatible = "aspeed,ast2600-spi";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		label = "pnor";
+		status = "okay";
+	};
+
+};
+
+&uart1 {
+	//Host console
+	compatible = "aspeed,ast2600-uart";
+	/delete-property/ reg-shift;
+	/delete-property/ reg-io-width;
+	dma-mode;
+	dma-channel = <0>;
+	status = "okay";
+};
+
+//BMC Console
+&uart5 {
+	status = "okay";
+};
+
+// I2C configs
+&i2c0 {
+	//Net name i2c9 - MB FPGA slave
+	status = "okay";
+};
+
+&i2c1 {
+	//Net name i2c10 - galena side i2c12
+	// Connected to TCA9546A and to LCD HDR
+	status = "okay";
+#ifdef OPENBMC_LCD_DRIVER
+	tca9546a@70 {
+		compatible = "nxp,pca9546";
+		reg = <0x70>;
+		//<TBD> Define LCD details here
+	};
+#endif //OPENBMC_LCD_DRIVER
+};
+
+&i2c4 {
+	// Net name i2c1
+	status = "okay";
+};
+
+// BMC galena VRs
+&i2c5 {
+	// Net name i2c2 (SCM_I2C2)
+	status = "okay";
+
+	i2cswitch@71 {
+		compatible = "nxp,pca9546";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+			core0socvrm@40 {
+				//VDD core0/soc VRM
+				compatible = "infineon,xdpe12284";
+				reg = <0x40>;
+			};
+			iovrm@41 {
+				//VDD io VRM
+				compatible = "infineon,xdpe12284";
+				reg = <0x41>;
+			};
+			core1vrm@42 {
+				//VDD core1 VRM
+				compatible = "infineon,xdpe12284";
+				reg = <0x42>;
+			};
+			vdd11susvrm@70 {
+				//VDD core1 VRM
+				compatible = "infineon,xdpe12284";
+				reg = <0x70>;
+			};
+			vdd33dualvrm@16 {
+				//VDD 33 DUAL VRM
+				compatible = "infineon,xdpe12284";
+				reg = <0x16>;
+			};
+			vdd18dualvrm@17 {
+				//VDD 18 DUAL VRM
+				compatible = "infineon,xdpe12284";
+				reg = <0x17>;
+			};
+		};
+	};
+};
+
+&i2c7 {
+	// Net name i2c4
+	status = "okay";
+
+	mbeeprom@50 {
+	compatible = "microchip,24lc256","atmel,24c256";
+	reg = <0x50>;
+	};
+};
+
+&i2c8 {
+	// Net name i2c5 - PSU
+	status = "okay";
+};
+
+&i2c10 {
+	// Net name i2c7
+	status = "okay";
+
+#ifdef EEPROM_PROG_ENABLE
+	i2cswitch@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+
+	};
+
+	i2cswitch@71 {
+		compatible = "nxp,pca9546";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+
+			brdreveeprom@50 {
+				//Galena BRD ID
+				compatible = "microchip,24lc256","atmel,24c256";
+				reg = <0x50>;
+			};
+
+			scmbrdeeprom@51 {
+				//SCM BRD ID
+				compatible = "microchip,24lc256","atmel,24c256";
+				reg = <0x51>;
+			};
+		};
+
+	};
+#endif
+};
+
+&i2c11 {
+	// Net name i2c8 - MB LOM
+	status = "okay";
+};
+
+&i2c12 {
+	// LOCAL MGMT - FPGA slave
+	status = "okay";
+};
+
+&i2c14 {
+	// LOCAL MGMT - EEPROM, RTC
+	status = "okay";
+	bmceeprom@50 {
+	compatible = "microchip,24lc512","atmel,24c512";
+	reg = <0x50>;
+	};
+};
+
+&i2c15 {
+	// RoT
+	status = "okay";
+};
+
+&i2c9 {
+	// Net name i2c6
+	status = "okay";
+
+	i2cswitch@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		//  port 0 to 3 connected to 4 fan controllers which controls 5 fans each
+		i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@5 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <5>;
+
+			lm75a@48 {
+				compatible = "national,lm75a";
+				reg = <0x48>;
+			};
+
+			lm75a@49 {
+				compatible = "national,lm75a";
+				reg = <0x49>;
+			};
+
+			lm75a@4a {
+				compatible = "national,lm75a";
+				reg = <0x4a>;
+			};
+
+			lm75a@4b {
+				compatible = "national,lm75a";
+				reg = <0x4b>;
+			};
+
+			lm75a@4c {
+				compatible = "national,lm75a";
+				reg = <0x4c>;
+			};
+
+			lm75a@4d {
+				compatible = "national,lm75a";
+				reg = <0x4d>;
+			};
+
+			lm75a@4e {
+				compatible = "national,lm75a";
+				reg = <0x4e>;
+			};
+
+			lm75a@4f {
+				compatible = "national,lm75a";
+				reg = <0x4f>;
+			};
+		};
+
+		i2c@6 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <6>;
+
+			adc0: fpga_vdd_core@50 {
+				compatible = "ti,adc121c";
+				reg = <0x50>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_core";
+			};
+			adc1: fpga_vdd_18@51 {
+				compatible = "ti,adc121c";
+				reg = <0x51>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_18";
+			};
+			adc2: fpga_vdd_33@52 {
+				compatible = "ti,adc121c";
+				reg = <0x52>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_33";
+			};
+			adc3: lom_vdd_33@53 {
+				compatible = "ti,adc121c";
+				reg = <0x53>;
+				#io-channel-cells = <1>;
+				label = "lom_vdd_33";
+			};
+			adc4: lom_vdd_12@54 {
+				compatible = "ti,adc121c";
+				reg = <0x54>;
+				#io-channel-cells = <1>;
+				label = "lom_vdd_12";
+			};
+			adc5: vdd_12_dual@55 {
+				compatible = "ti,adc121c";
+				reg = <0x55>;
+				#io-channel-cells = <1>;
+				label = "vdd_12_dual";
+			};
+		};
+
+		i2c@7 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <7>;
+
+			tmp468@48 {
+				compatible = "ti,tmp468";
+				reg = <0x48>;
+			};
+		};
+	};
+};
+
+// i3c
+
+&i3c0 {
+	// P0 DIMM (A-F) SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x00>;
+
+	//DIMM A
+	spd5118_0_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_0_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_0_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM B
+	spd5118_0_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_0_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_0_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM C
+	spd5118_0_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_0_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_0_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+	//DIMM D
+	spd5118_0_3: spd@53,3C000000003 {
+		reg = <0x53 0x3C0 0x00000003>;
+		assigned-address = <0x53>;
+	};
+	pmic5xxx_0_3: pmic@4b,20400000003 {
+		reg = <0x4b 0x204 0x00000003>;
+		assigned-address = <0x4b>;
+	};
+	rcd_0_3: rcd@5b,2C000000003 {
+		reg = <0x5b 0x2C0 0x00000003>;
+		assigned-address = <0x5b>;
+	};
+	//DIMM E
+	spd5118_0_4: spd@54,3C000000004 {
+		reg = <0x54 0x3C0 0x00000004>;
+		assigned-address = <0x54>;
+	};
+	pmic5xxx_0_4: pmic@4c,20400000004 {
+		reg = <0x4c 0x204 0x00000004>;
+		assigned-address = <0x4c>;
+	};
+	rcd_0_4: rcd@5c,2C000000004 {
+		reg = <0x5c 0x2C0 0x00000004>;
+		assigned-address = <0x5c>;
+	};
+	//DIMM F
+	spd5118_0_5: spd@55,3C000000005 {
+		reg = <0x55 0x3C0 0x00000005>;
+		assigned-address = <0x55>;
+	};
+	pmic5xxx_0_5: pmic@4d,20400000005 {
+		reg = <0x4d 0x204 0x00000005>;
+		assigned-address = <0x4d>;
+	};
+	rcd_0_5: rcd@5d,2C000000005 {
+		reg = <0x5d 0x2C0 0x00000005>;
+		assigned-address = <0x5d>;
+	};
+};
+
+&i3c1 {
+	// P0 DIMM (G-L) SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x01>;
+
+	//DIMM G
+	spd5118_1_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_1_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_1_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM H
+	spd5118_1_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_1_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_1_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM I
+	spd5118_1_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_1_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_1_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+	//DIMM J
+	spd5118_1_3: spd@53,3C000000003 {
+		reg = <0x53 0x3C0 0x00000003>;
+		assigned-address = <0x53>;
+	};
+	pmic5xxx_1_3: pmic@4b,20400000003 {
+		reg = <0x4b 0x204 0x00000003>;
+		assigned-address = <0x4b>;
+	};
+	rcd_1_3: rcd@5b,2C000000003 {
+		reg = <0x5b 0x2C0 0x00000003>;
+		assigned-address = <0x5b>;
+	};
+	//DIMM K
+	spd5118_1_4: spd@54,3C000000004 {
+		reg = <0x54 0x3C0 0x00000004>;
+		assigned-address = <0x54>;
+	};
+	pmic5xxx_1_4: pmic@4c,20400000004 {
+		reg = <0x4c 0x204 0x00000004>;
+		assigned-address = <0x4c>;
+	};
+	rcd_1_4: rcd@5c,2C000000004 {
+		reg = <0x5c 0x2C0 0x00000004>;
+		assigned-address = <0x5c>;
+	};
+	//DIMM L
+	spd5118_1_5: spd@55,3C000000005 {
+		reg = <0x55 0x3C0 0x00000005>;
+		assigned-address = <0x55>;
+	};
+	pmic5xxx_1_5: pmic@4d,20400000005 {
+		reg = <0x4d 0x204 0x00000005>;
+		assigned-address = <0x4d>;
+	};
+	rcd_1_5: rcd@5d,2C000000005 {
+		reg = <0x5d 0x2C0 0x00000005>;
+		assigned-address = <0x5d>;
+	};
+};
+
+#ifdef ENABLE_I2C_APML
+
+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbrmi@3c {
+	compatible = "amd,sbrmi";
+		reg = <0x3c>;
+	};
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+};
+
+#else
+
+&i3c4 {
+	// P0 APML
+	status = "okay";
+
+	jdec-spd;
+	set_dasa;
+	bus_id = <0x04>;
+
+	imx3112_p0: i3cmux@70,4CC00000000 {
+		reg = <0x70 0x4CC 0x00000000>;
+		assigned-address = <0x70>;
+	};
+	sbtsi_p0_0: sbtsi@4c,22400000001 {
+		reg = <0x4c 0x224 0x00000001>;
+		assigned-address = <0x4c>;
+	};
+	sbrmi_p0_1: sbrmi@3c,22400000002 {
+		reg = <0x3c 0x224 0x00000002>;
+		assigned-address = <0x3c>;
+	};
+};
+
+#endif  // ENABLE_I2C_APML
+
+&espi_ctrl {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_espi_default &pinctrl_espialt_default>;
+
+	perif,dma-mode;
+};
+
+&espi_mmbi {
+	status = "okay";
+};
+
+&gpio0 {
+	gpio-line-names =
+	/*A0-A7*/	"","","","","","","","",
+	/*B0-B7*/	"","","","","MON_POST_COMPLETE","","","",
+	/*C0-C7*/	"","","","","","","","",
+	/*D0-D7*/	"","","","","","","","",
+	/*E0-E7*/	"","","","","","","","",
+	/*F0-F7*/	"","","","","","","","",
+	/*G0-G7*/	"","","","","","","","",
+	/*H0-H7*/	"","ASSERT_WARM_RST_BTN_L","ASSERT_SOC_RST_BTN_L","","","","","",
+	/*I0-I7*/	"","","","","","","","P0_I3C_APML_ALERT_L",
+	/*J0-J7*/	"","","","","","","","",
+	/*K0-K7*/	"","","","","","","","",
+	/*L0-L7*/	"","","","","","","","",
+	/*M0-M7*/	"","","","","","","","",
+	/*N0-N7*/	"","","","","","","PSP_SOFT_FUSE_NOTIFY","ASSERT_BMC_READY",
+	/*O0-O7*/	"","","HDT_SEL","HDT_XTRIG5","HDT_XTRIG6","JTAG_TRST_N","","",
+	/*P0-P7*/	"MON_RST_BTN_L","ASSERT_RST_BTN_L","MON_PWR_BTN_L","ASSERT_PWR_BTN_L","","ASSERT_NMI_BTN_L","MON_PWR_GOOD","",
+	/*Q0-Q7*/	"","","HDT_DBREQ_L","","BIOS_SPD_MUX_CTRL_RELEASED_L","","","",
+	/*R0-R7*/	"","","","","","","","",
+	/*S0-S7*/	"","","","","","","P0_DIMM_AF_ERROR","P0_DIMM_GL_ERROR",
+	/*T0-T7*/	"","","","","","","","",
+	/*U0-U7*/	"","","","","","","","",
+	/*V0-V7*/	"","","","","","","","",
+	/*W0-W7*/	"","","","","","","","",
+	/*X0-X7*/	"","","","","","","","",
+	/*Y0-Y7*/	"","","","","","","","",
+	/*Z0-Z7*/	"","","","","","","","";
+};
+
+&jtag1 {
+	status = "okay";
+};
+
+&video {
+	status = "okay";
+	memory-region = <&video_engine_memory>;
+};
+
+&vhub {
+	status = "okay";
+};
+
+&lpc_ctrl {
+	status = "okay";
+};
+
+&lpc_pcc {
+	port-addr = <0x80>;
+	dma-mode;
+	memory-region = <&pcc_memory>;
+	rec-mode = <0x1>;
+	port-addr-hbits-select = <0x1>;
+	port-addr-xbits = <0x3>;
+
+	status = "okay";
+};
+
+&kcs3 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xCA2>;
+};
+
+// eSPI Virtual UART - 0x3F8
+&vuart1 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts b/arch/arm/boot/dts/aspeed-bmc-amd-huambo.dts
similarity index 100%
rename from arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
rename to arch/arm/boot/dts/aspeed-bmc-amd-huambo.dts
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-purico.dts
similarity index 100%
rename from arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
rename to arch/arm/boot/dts/aspeed-bmc-amd-purico.dts
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-recluse.dts
similarity index 100%
rename from arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
rename to arch/arm/boot/dts/aspeed-bmc-amd-recluse.dts
--
2.25.1
